lib/tests/streamlit/legacy_dataframe_styling_test.py

```diff
@@ -14,62 +14,101 @@
         """
 
-        df = pd.DataFrame({"A": [1, 2, 3, 4, 5]})
+        values = [1, 2, 3, 4, 5]
+        display_values = [None] * 5
+        df = pd.DataFrame({"A": values})
 
-        getattr(st, element)(df.style)
-        proto_df = getattr(self._get_element(), proto)
+        st_element(df)
+        proto_df = get_proto(self._get_element())
+        self._assert_column_display_values(proto_df, 0, display_values)
 
-        rows, cols = df.shape
-        for row in range(rows):
-            for col in range(cols):
-                style = get_cell_style(proto_df, col, row)
-                self.assertEqual(style.display_value, "")
-                self.assertEqual(style.has_display_value, False)
-                self.assertEqual(len(style.css), 0)
+    @parameterized.expand(
+        [(st._legacy_dataframe, _get_df_proto), (st._legacy_table, _get_table_proto)]
+    )
+    def test_default_style_has_style_data(self, st_element, get_proto):
+        """A DataFrame with a default Styler will have styling data."""
+
+        values = [1, 2, 3, 4, 5]
+        display_values = ["1", "2", "3", "4", "5"]
+        df = pd.DataFrame({"A": values})
+
+        st_element(df.style)
+        proto_df = get_proto(self._get_element())
+        self._assert_column_display_values(proto_df, 0, display_values)
 
     @parameterized.expand(
-        [("_legacy_dataframe", "data_frame"), ("_legacy_table", "table")]
+        [(st._legacy_dataframe, _get_df_proto), (st._legacy_table, _get_table_proto)]
     )
-    def test_format(self, element, proto):
+    def test_format_percent(self, st_element, get_proto):
         """Tests DataFrame.style.format()"""
         values = [0.1, 0.2, 0.3352, np.nan]
         display_values = ["10.00%", "20.00%", "33.52%", "nan%"]
 
         df = pd.DataFrame({"A": values})
 
-        get_delta = getattr(st, element)
-        get_delta(df.style.format("{:.2%}"))
+        st_element(df.style.format("{:.2%}"))
+
+        proto_df = get_proto(self._get_element())
+        self._assert_column_display_values(proto_df, 0, display_values)
+
+    @parameterized.expand(
+        [(st._legacy_dataframe, _get_df_proto), (st._legacy_table, _get_table_proto)]
+    )
+    def test_format_float_precision(self, st_element, get_proto):
+        """Tests DataFrame.style.format() with floats.
+        By default, the frontend will format any unstyled DataFrame float
+        with 4 digits after the decimal. If we have any floating point styling
+        in a DataFrame, our display_values should be filled in even for
+        cells whose display_value == value.
+        """
+        values = [3.14, 3.1]
+        display_values = ["3.14", "3.10"]
+
+        df = pd.DataFrame({"test": values})
+
+        st_element(df.style.format({"test": "{:.2f}"}))
 
-        proto_df = getattr(self._get_element(), proto)
+        proto_df = get_proto(self._get_element())
         self._assert_column_display_values(proto_df, 0, display_values)
```

Comment:
I added this test while dealing with e2e test failures. It uses the same data as the `st_legacy_dataframe_format.py` e2e test; it's useful to have it here to catch changes that would also result in an e2e failure. (The e2e test is still important because it asserts some frontend rendering logic.)